pkg/util/wsstream/conn.go

```diff
@@ -123,21 +124,28 @@ func handshake(config *websocket.Config, req *http.Request, allowed []string) er
 // Conn supports sending multiple binary channels over a websocket connection.
 // Supports only the "channel.k8s.io" subprotocol.
 type Conn struct {
-	channels []*websocketChannel
-	codec    codecType
-	ready    chan struct{}
-	ws       *websocket.Conn
-	timeout  time.Duration
+	channels           []*websocketChannel
+	codec              codecType
+	ready              chan struct{}
+	ws                 *websocket.Conn
+	timeout            time.Duration
+	supportedProtocols []string
+	protocolVersion    string
 }
 
 // NewConn creates a WebSocket connection that supports a set of channels. Channels begin each
 // web socket message with a single byte indicating the channel number (0-N). 255 is reserved for
 // future use. The channel types for each channel are passed as an array, supporting the different
 // duplex modes. Read and Write refer to whether the channel can be used as a Reader or Writer.
-func NewConn(channels ...ChannelType) *Conn {
+//
+// Protocols have the shape "channel.k8s.io" in the unversioned case and "channel.k8s.io/v4" in
```

Comment:
I am fine with the dot notation, will change that. I guess a `protocolVersion` becomes `protocolPrefix` then. Can you comment on the `Conn` API? Is that ok as it is (with the renamings)? I would like to avoid the two protocols `channel.k8s.io` and `base64.channel.k8s.io` leaking outside of this package.